vector/src/main/java/fr/gouv/tchap/features/login/registration/TchapRegisterFragment.kt

```diff
@@ -78,17 +79,25 @@ class TchapRegisterFragment @Inject constructor(private val platformViewModelFac
 
         loginViewModel.observeViewEvents { loginViewEvents ->
             when (loginViewEvents) {
-                TchapLoginViewEvents.OnLoginFlowRetrieved -> loginViewModel.handle(TchapLoginAction.LoginOrRegister(login, password, getString(R.string.login_default_session_public_name)))
-                is TchapLoginViewEvents.RegistrationFlowResult -> {
+                TchapLoginViewEvents.OnLoginFlowRetrieved       -> loginViewModel.handle(TchapLoginAction.LoginOrRegister(login, password, getString(R.string.login_default_session_public_name)))
+                is TchapLoginViewEvents.RegistrationFlowResult  -> {
+                    // Result from registration request when the account password is set.
+                    // Email stage is mandatory at this time and another stage should not happen.
                     val emailStage = loginViewEvents.flowResult.missingStages.firstOrNull { it.mandatory && it is Stage.Email }
 
                     if (emailStage != null) {
                         loginViewModel.handle(TchapLoginAction.AddThreePid(RegisterThreePid.Email(login)))
-                    } else Unit
+                    } else {
+                        AlertDialog.Builder(requireActivity())
+                                .setTitle(R.string.dialog_title_error)
+                                .setMessage(R.string.login_error_unable_register)
+                                .setPositiveButton(R.string.ok, null)
+                                .show()
+                        Unit
```

Comment:
Don't you think we can remove this Unit?

Reply:
No it's normal